tcms/core/tests/test_forms.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=invalid-name
+
+from django.test import TestCase
+
+from tcms.core.forms.fields import UserField
+from tcms.tests.factories import UserFactory
+from django.core.exceptions import ValidationError
+
+
+class TestUserField(TestCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.user = UserFactory(username='admin', email='admin@example.com')
+        cls.user.set_password('admin')  # nosec:B106:hardcoded_password_funcarg
+        cls.user.is_superuser = True
+        cls.user.is_staff = True  # enables access to admin panel
+        cls.user.save()
+
+        cls.userField = UserField()
+
+    def test_valid_user(self):
+        formUser = self.userField.clean("admin")
+        self.assertEqual(formUser.email, self.user.email)
+        self.assertEqual(formUser.username, self.user.username)
+
+    def test_empty_username(self):
+        with self.assertRaisesRegex(ValidationError, 'A user name or user ID is required.'):
+            self.userField.clean('')
+
+    def test_none_username(self):
+        with self.assertRaisesRegex(ValidationError, 'A user name or user ID is required.'):
+            self.userField.clean(None)
+
+    def test_none_username_not_required(self):
+        self.userField.required = False
+        self.assertIsNone(self.userField.clean(None))
+        self.userField.required = True
+
+    def test_empty_username_not_required(self):
+        self.userField.required = False
+        self.assertIsNone(self.userField.clean(''))
+        self.userField.required = True
+
+    def test_int_instance_username(self):
+        userID = int(self.user.id)
+        formUser = self.userField.clean(userID)
```

Comment:
You should use camel_case for these variables, also no need for the extra variable. We use pk instead of id in most cases.

Reply:
Without a variable it breaks the max-line-length rule. IMO it should be kept as variable. 
What do you think about "user_id = self.user.pk"?

Reply:
FTR `thisIsCamelCase` but `python_prefers_underscores`.

Reply:
> Without a variable it breaks the max-line-length rule.

It definitely doesn't, see the status report from CI. You are using different configuration locally, see Makefile.

Reply:
yeah, srry my bad,I meant snake_case